prov/gni/src/gnix_sep.c

```diff
@@ -886,6 +886,146 @@ gnix_sep_msg_injectdata(struct fid_ep *ep, const void *buf, size_t len,
 			  NULL, flags, data, 0);
 }
 
+DIRECT_FN STATIC ssize_t gnix_sep_trecv(struct fid_ep *ep, void *buf,
+					size_t len,
+					void *desc, fi_addr_t src_addr,
+					uint64_t tag, uint64_t ignore,
+					void *context)
+{
+	struct gnix_fid_trx *rx_ep = container_of(ep, struct gnix_fid_trx,
+						  ep_fid);
+
+	return _ep_recv(&rx_ep->ep->ep_fid, buf, len, desc, src_addr, context,
+			FI_TAGGED, tag, ignore);
+}
+
+DIRECT_FN STATIC ssize_t gnix_sep_trecvv(struct fid_ep *ep,
+					 const struct iovec *iov,
+					 void **desc, size_t count,
+					 fi_addr_t src_addr,
+					 uint64_t tag, uint64_t ignore,
+					 void *context)
+{
+	struct gnix_fid_trx *rx_ep = container_of(ep, struct gnix_fid_trx,
+						  ep_fid);
+
+	return _ep_recvv(&rx_ep->ep->ep_fid, iov, desc, count, src_addr,
+			 context, FI_TAGGED, tag, ignore);
+}
+
+DIRECT_FN STATIC ssize_t gnix_sep_trecvmsg(struct fid_ep *ep,
+					   const struct fi_msg_tagged *msg,
+					   uint64_t flags)
+{
+	uint64_t clean_flags;
+
+	const struct fi_msg _msg = {
+			.msg_iov = msg->msg_iov,
+			.desc = msg->desc,
+			.iov_count = msg->iov_count,
+			.addr = msg->addr,
+			.context = msg->context,
+			.data = msg->data
+	};
+
+	clean_flags = (flags & (GNIX_TRECVMSG_FLAGS)) | FI_TAGGED;
```

Comment:
@ztiffany : Yes, I understand that we are preventing garbage flags from being passed into the library. It's more of a matter of saying 'Should we allow this to look like a valid call, when obviously the application is passing garbage flag? If the flags are garbage, but the user is expecting a certain type of action, then if we complete the call, is it reasonable for the user to expect that action to be complete?'.

I think the answer to that question is 'Yes, because we allowed to call to happen and return FI_SUCCESS'.

Yes, we should be checking the following

```C
if (flags & ~GNIX_TRECVMSG_FLAGS)
    return -FI_EINVAL;
```

or some form of the same check... it doesn't matter what it looks like so much as it is functional and makes sense.